### PR TITLE
Fix upgrade behaviors

### PR DESCRIPTION
**Pull request by mhofman**
## Description
- Upgrades Zoe and repair KREAd
  - KREAd proposal and tests are a port from the release branch version used during upgrade 14
- restores the old probe zcf test behavior now that https://github.com/Agoric/agoric-sdk/issues/9249 was fixed in u15
- Includes https://github.com/Agoric/agoric-sdk/pull/9536

### Security Considerations
None

### Scaling Considerations
None

### Documentation Considerations
None

### Testing Considerations
Restores previously disabled or omited tests

### Upgrade Considerations
Change to core proposal in the next software upgrade (upgrade 16)

### CODE REVIEW

File: a3p-integration/proposals/a:upgrade-next/upgradeVaults.test.js [hunk 1]
```diff
@@ -71,7 +54,7 @@ const tryPushPrices = async t => {
   const osmoOut = await getPriceQuote('stOSMO');
   t.is(osmoOut, '+11500000');
   const stkAtomOut = await getPriceQuote('stkATOM');
-  t.is(stkAtomOut, '+11600000');
+  await t.is(stkAtomOut, '+11600000');
```
Comment (turadg) [inline a3p-integration/proposals/a:upgrade-next/upgradeVaults.test.js]: Noop, remove

File: golang/cosmos/app/app.go [hunk 2]
```diff
@@ -934,8 +934,10 @@ func unreleasedUpgradeHandler(app *GaiaApp, targetUpgrade string) func(sdk.Conte
 			// Each CoreProposalStep runs sequentially, and can be constructed from
 			// one or more modules executing in parallel within the step.
 			CoreProposalSteps = []vm.CoreProposalStep{
-				// Upgrade ZCF only
-				vm.CoreProposalStepForModules("@agoric/builders/scripts/vats/upgrade-zcf.js"),
+				// Upgrade Zoe + ZCF
+				vm.CoreProposalStepForModules("@agoric/builders/scripts/vats/replace-zoe.js"),
+				// Revive KREAd characters
+				vm.CoreProposalStepForModules("@agoric/builders/scripts/vats/revive-kread.js"),
```
Comment (turadg) [inline golang/cosmos/app/app.go:940]: I'm not privy to the plan so I'm just assuming this is supposed to be done

File: packages/vats/src/proposals/kread-proposal.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+/**
+ * @param {BootstrapPowers & {
+ *   consume: { kreadKit: any };
+ * }} powers
+ */
+export const repairKread = async ({ consume: { kreadKit: kreadKitP } }) => {
+  console.log('repairSubscribers');
+
+  const kreadKit = await kreadKitP;
+  const creatorFacet = kreadKit.creatorFacet;
```
Comment (turadg) [inline packages/vats/src/proposals/kread-proposal.js:12]: Consider destrucuring. Could be directly from the promise as 'kreadKit' is unused

**Review by turadg — APPROVED** [review]

File: packages/vats/src/proposals/kread-proposal.js (hunk 3 above)
Comment (mhofman) [reply packages/vats/src/proposals/kread-proposal.js:12]: This is a direct checkout of the `release-mainnet1B` file

File: golang/cosmos/app/app.go (hunk 2 above)
Comment (mhofman) [reply golang/cosmos/app/app.go:940]: Yes, Zoe needs to be updated, and that forces us to repair KREAd

File: a3p-integration/proposals/a:upgrade-next/upgradeVaults.test.js (hunk 1 above)
Comment (mhofman) [reply a3p-integration/proposals/a:upgrade-next/upgradeVaults.test.js]: This is straight cherry-pick of https://github.com/Agoric/agoric-sdk/pull/9536, arguably I should have reviewed that better

**Review by ivanlei — APPROVED** [review]
I reviewed only app.go

**Pull request merged**